app/src/marketplace/containers/EditProductPage2/DeployCommunityModal.jsx

```diff
@@ -34,14 +35,16 @@ function setSkipGuide(value) {
 export const DeployDialog = ({ product, api }: DeployDialogProps) => {
     const dontShowAgain = skipGuide()
     const [step, setStep] = useState(dontShowAgain ? steps.CONFIRM : steps.GUIDE)
+    const { deployContract } = useContext(EditControllerContext)
 
     const onClose = useCallback(() => {
         api.close(false)
     }, [api])
 
-    const onDeploy = useCallback(() => {
+    const onDeploy = useCallback(async () => {
+        await deployContract()
         setStep(steps.COMPLETE)
```

Comment:
The process here could be:
- set step to `COMPLETE`
- create the stream
- Add e&e write permission and public permission, remove own share permission
- Calculate address and set to dialog result (exiting at this phase signifies success)
- Start deploy & show spinner